### PR TITLE
Feat/landing typology filter, closes issue #57

### PR DESCRIPTION
**Pull request by creativogee**
## Description

This commit contains:
- Responsive component with which a marketplace user could filter asset list: Filter options include virtual land typologies
- Minor refactor to the header to follow design changes

This commit does not contain:
- Functionality

### CODE REVIEW

**Comment by anjorinjnr** [comment]
I am confused about this UI?

Is this the typology filter?

image

What is a user supposed to type in the search box?

Also the drop down listsour land types? That's very restrictive. That dropdown should probably be asset types (LAND, ART, etc)

**Review by anjorinjnr — CHANGES_REQUESTED** [review]
see my comment

**Comment by anjorinjnr** [comment]
And shouldn't close until they are functional unless we plan to or have another issue for implementing the functionality

**Comment by creativogee** [comment]
This part is
image
What you highlighted is the search component. I believe the idea behind that is to have a finer-grained search like searching by name. Also, my guess is those options are like that in the dropdown because VLAND is the only asset for now.

**Comment by creativogee** [comment]
> And shouldn't close until they are functional unless we plan to or have another issue for implementing the functionality

Actually most tickets, for now, say to just build it out. We could create new tickets to implement the functionality, we could as well wait until functionality is implemented

**Comment by jmulq** [comment]
> > And shouldn't close until they are functional unless we plan to or have another issue for implementing the functionality
> 
> Actually most tickets, for now, say to just build it out. We could create new tickets to implement the functionality, we could as well wait until functionality is implemented

The build out tickets are for building out the functionality/logic as well as the UI. If we're creating new components that don't implement any of the logic, lets call them scaffold. Also could they be tagged appropriately.

File: src/features/elements/TypologyFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  return (
+    <div className='pb-5 sm:border-b border-gray-200 sm:pb-0 mb-6'>
+      <nav className='flex sm:space-x-8 justify-center flex-col sm:flex-row items-center'>
+        {searchOptions.map((tab) => (
+          <a
+            key={tab}
+            onClick={() => handleClick(tab)}
+            className={`whitespace-nowrap pb-2 mb-2 md:mb-0 md:pb-4 px-1 border-b-2 font-medium text-sm cursor-pointer
```
Comment (jmulq) [inline src/features/elements/TypologyFilter.tsx:18]: Have you heard of the `classnames` package? Is a cleaner way of doing this. Might be worth implementing it across the project https://www.npmjs.com/package/classnames.

File: src/pages/home/index.tsx [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 import Promotion from '../../features/elements/Promotion'
+import TypologyFilter from '../../features/elements/TypologyFilter'
 import { AssetsListing } from '../../features/partials/AssetsListing'
 import DefaultLayoutContainer from '../../features/shared/layout/DefaultLayoutContainer'
 
 const Landing = () => {
   return (
     <DefaultLayoutContainer>
       <Hero />
+      <TypologyFilter />
```
Comment (jmulq) [inline src/pages/home/index.tsx:12]: How are we planning to change the assets passed down to the `AssetListing` component from the `TypologyFilter`?  Just wondering

Comment (creativogee) [reply src/pages/home/index.tsx:12]: > How are we planning to change the assets passed down to the `AssetListing` component from the `TypologyFilter`? Just wondering

We could use the store to achieve that

File: src/features/elements/TypologyFilter.tsx (hunk 1 above)
Comment (creativogee) [reply src/features/elements/TypologyFilter.tsx:18]: > Have you heard of the `classnames` package? Is a cleaner way of doing this. Might be worth implementing it across the project https://www.npmjs.com/package/classnames.

Alright noted

**Comment by creativogee** [comment]
> > > And shouldn't close until they are functional unless we plan to or have another issue for implementing the functionality
> > 
> > 
> > Actually most tickets, for now, say to just build it out. We could create new tickets to implement the functionality, we could as well wait until functionality is implemented
> 
> The build out tickets are for building out the functionality/logic as well as the UI. If we're creating new components that don't implement any of the logic, lets call them scaffold. Also could they be tagged appropriately.

That makes sense, at the time there was no server to make requests to and the store was not set up. Also, this current ticket did not have a dependency on a server ticket. So, should I create a new ticket that says scaffold while I probably focus on setting up the redux store on a separate?

File: src/features/elements/TypologyFilter.tsx (hunk 1 above)
Comment (anjorinjnr) [reply src/features/elements/TypologyFilter.tsx:18]: this package is 20kb, do we need it?

A util function like this is sufficient
```
export const classNames = (...classes: string[]) =>
  classes.filter(Boolean).join(' ')
```
you can do something like 
```
<CheckIcon
    className={classNames(
       pctCompleted === 100 ? 'text-green-600' : 'text-gray-500',
       'h-5 w-5'
       )}
 />
```

**Comment by anjorinjnr** [comment]
> This part is image What you highlighted is the search component. I believe the idea behind that is to have a finer-grained search like searching by name. Also, my guess is those options are like that in the dropdown because VLAND is the only asset for now.

This doesn't generalize. Unless we are building a vland specific page.

We should keep in that vland are not the only thing that will be traded in the markerplace.

I thought we can building the side filter, which we can generate from the asset metadata and will be dynamic.

**Comment by anjorinjnr** [comment]
> > > And shouldn't close until they are functional unless we plan to or have another issue for implementing the functionality
> > 
> > 
> > Actually most tickets, for now, say to just build it out. We could create new tickets to implement the functionality, we could as well wait until functionality is implemented
> 
> The build out tickets are for building out the functionality/logic as well as the UI. If we're creating new components that don't implement any of the logic, lets call them scaffold. Also could they be tagged appropriately.

Either option is fine for me, make be for planning purposed might be easier to use different issues. Just name and tag them propertly.

@jmulq  @abbeydabiri  fyi

**Comment by anjorinjnr** [comment]
> I am confused about this UI?
> 
> Is this the typology filter?
> 
> image
> 
> What is a user supposed to type in the search box?
> 
> Also the drop down lists our land types? That's very restrictive. That dropdown should probably be asset types (LAND, ART, etc)

we should remove search for now.

File: src/features/elements/TypologyFilter.tsx (hunk 1 above)
Comment (creativogee) [reply src/features/elements/TypologyFilter.tsx:18]: > this package is 20kb, do we need it?
> 
> A util function like this is sufficient
> 
> ```
> export const classNames = (...classes: string[]) =>
>   classes.filter(Boolean).join(' ')
> ```
> 
> you can do something like
> 
> ```
> <CheckIcon
>     className={classNames(
>        pctCompleted === 100 ? 'text-green-600' : 'text-gray-500',
>        'h-5 w-5'
>        )}
>  />
> ```

agreed, less bundle size

**Pull request closed without merging**